lib/spack/spack/test/mirror.py

```diff
@@ -102,6 +102,7 @@ def check_mirror(self):
                 spec = Spec(name).concretized()
                 pkg = spec.package
 
+                pkg._stage = None
```

Comment:
@tgamblin : tests started failing here after merge with develop. The cause of failure was that the stage directory for the package does not exist at this point in the code. 

This issue stems from two concurrent causes : 
 1. the stage directory is created in the "private" method `Stage._setup` called only during `Stage.__init__` while the same directory can be destroyed via a "public" method `Stage.destroy`
 2. the `Package` object used here is the same used to create the mirror. This means that the stage directory has been created (and destroyed) a few lines above, and that here we expect to be able to `chdir` to a non-existing place

I temporarily solved the issue in the test by forcing the creation of a new `Stage` object, but I don't think this will be a good solution in the long run. What do you think about turning the `Stage` class into a context manager (either in this PR or in a separate one)?